pkg/tools/sandbox/builder.go

```diff
@@ -281,191 +149,175 @@ func (b *Builder) SetRegistryExpiryDuration(registryExpiryDuration time.Duration
+func (b *Builder) UseUnixSockets() *Builder {
+	require.NotEqual(b.t, "windows", runtime.GOOS, "Unix sockets are not available for windows")
 
-	serve(ctx, serveURL, mgr.Register)
-	log.FromContext(ctx).Debugf("%v: %v listen on: %v", b.DNSDomainName, name, serveURL)
-	return &EndpointEntry{
-		Endpoint: mgr,
-		URL:      serveURL,
-	}
+	b.nodesCount = 1
+	b.supplyNSMgrProxy = nil
+	b.supplyRegistry = nil
+	b.supplyRegistryProxy = nil
+	b.useUnixSockets = true
+	return b
 }
 
-// NewNSMgr - starts new Network Service Manager
-func (b *Builder) NewNSMgr(ctx context.Context, node *Node, address string, registryURL *url.URL, generateTokenFunc token.GeneratorFunc) (entry *NSMgrEntry, resources []context.CancelFunc) {
-	nsmgrCtx, nsmgrCancel := context.WithCancel(ctx)
-	b.resources = append(b.resources, nsmgrCancel)
-
-	entry = b.newNSMgr(nsmgrCtx, address, registryURL, generateTokenFunc)
+// Build builds Domain and Supplier
+func (b *Builder) Build() *Domain {
+	b.domain = &Domain{
+		Name:        b.name,
+		DNSResolver: b.dnsResolver,
+	}
 
-	b.SetupRegistryClients(ctx, node)
+	if b.useUnixSockets {
+		msg := "Unix sockets are available only for local tests with no external registry"
+		require.Equal(b.t, b.nodesCount, 1, msg)
+		require.Nil(b.t, b.supplyNSMgrProxy, msg)
+		require.Nil(b.t, b.supplyRegistry, msg)
+		require.Nil(b.t, b.supplyRegistryProxy, msg)
 
-	resources, b.resources = b.resources, nil
-	return
-}
+		sockPath, err := ioutil.TempDir(os.TempDir(), "nsm-domain-temp")
+		require.NoError(b.t, err)
 
-func (b *Builder) newNSMgr(ctx context.Context, address string, registryURL *url.URL, generateTokenFunc token.GeneratorFunc) *NSMgrEntry {
-	if b.supplyNSMgr == nil {
-		return nil
-	}
+		go func() {
+			<-b.ctx.Done()
+			_ = os.RemoveAll(sockPath)
+		}()
 
-	var serveURL *url.URL
-	if b.useUnixSockets {
-		serveURL = grpcutils.TargetToURL(address)
+		b.domain.supplyURL = b.supplyUnixAddress(sockPath, new(int))
 	} else {
-		listener, err := net.Listen("tcp", address)
-		b.require.NoError(err)
-		serveURL = grpcutils.AddressToURL(listener.Addr())
-		b.require.NoError(listener.Close())
+		b.domain.supplyURL = b.supplyTCPAddress()
 	}
 
-	nsmgrName := "nsmgr-" + uuid.New().String()
-
-	options := []nsmgr.Option{
-		nsmgr.WithName(nsmgrName),
-		nsmgr.WithAuthorizeServer(authorize.NewServer(authorize.Any())),
-		nsmgr.WithConnectOptions(
-			connect.WithDialTimeout(DialTimeout),
-			connect.WithDialOptions(DefaultDialOptions(generateTokenFunc)...)),
+	if b.supplyRegistryProxy != nil {
+		require.NotNil(b.t, b.supplyNSMgrProxy, "NSMgr proxy supplier should be set if registry proxy supplier is set")
+		b.domain.NSMgrProxy = &EndpointEntry{
+			URL: b.domain.supplyURL("nsmgr-proxy"),
+		}
 	}
 
-	if registryURL != nil {
-		options = append(options, nsmgr.WithRegistry(registryURL, DefaultDialOptions(generateTokenFunc)...))
+	b.newRegistryProxy()
+	b.newRegistry()
+	b.newNSMgrProxy()
```

Comment:
I think that it's a bit confusing when function with name `newX` changes state and doesn't return anything.
Maybe it would be better to use "setup", or "build", or "start", or something?

Reply:
Yep, fixed.